### PR TITLE
Feat/map display

### PR DESCRIPTION
**Pull request by joewilliams007**
Refined map, added debugging code, displaying of live statistics.

### CODE REVIEW

**Comment by joewilliams007** [comment]
I know someone had the same idea to create a Block model file. I guess this will be resolved during the merge?

**Comment by N3moAhead** [comment]
> I know someone had the same idea to create a Block model file. I guess this will be resolved during the merge?

Im guilty on that one XP

**Review by N3moAhead — COMMENTED** [review]
I have no objections, the code looks very good. I would like to merge the main branch into this one later.
After that it would be great if you @joewilliams007  @Driemtax could take another look at it. If everything looks good, we can then merge the pull request into the main branch. 😄

**Comment by N3moAhead** [comment]
Now we would have to check if everything still works as intended.
- The Block Enum now also has the Workplace Block...
- The Map width and height had been renamed to MAP_WIDTH and MAP_HEIGHT and are implemented as static final...


I don't think we have to connect frontend and backend completly in this pr. I would use a seperate PR to achieve that. 😄

**Comment by joewilliams007** [comment]
> I have no objections, the code looks very good. I would like to merge the main branch into this one later. After that it would be great if you @joewilliams007 @Driemtax could take another look at it. If everything looks good, we can then merge the pull request into the main branch. 😄

@Driemtax mentioned that the map should be responsive to the frame size. Which i am implementing at the moment.

File: src/com/biocollapse/view/MapPanel.java [hunk 1]
```diff
@@ -1,25 +1,188 @@
...
+import src.com.biocollapse.model.Block;
+import src.com.biocollapse.model.Human;
+import src.com.biocollapse.model.MapPosition;
+import src.com.biocollapse.util.GlobalRandom;
 
 public class MapPanel extends JPanel {
 
+    private static final int CELL_SIZE = 8;
+    private static int width = 100;
+    private static int height = 100;
```
Comment (N3moAhead) [inline src/com/biocollapse/view/MapPanel.java:27]: After this merge you can get the Map width/height from the Map model. 👍🏻

File: src/com/biocollapse/view/MapPanel.java [hunk 2]
```diff
@@ -1,25 +1,188 @@
...
+    private Block[][] map;
+    private List<Human> humanData;
+    private java.util.Map<Block, Color> legend;
+
+    private static final Color COLOR_INFECTED = Color.RED;
+    private static final Color COLOR_HEALTHY = Color.MAGENTA;
+
+    private static final String STATUS_INFECTED = "Infected";
+    private static final String STATUS_HEALTHY = "Healthy";
+
```
Comment (N3moAhead) [inline src/com/biocollapse/view/MapPanel.java:37]: I think we will also have to implement a color for the new Workplace Block, 🤔

File: src/com/biocollapse/view/MapPanel.java [hunk 3]
```diff
@@ -1,25 +1,188 @@
...
+        return legendPanel;
+    }
+
+    private void initLegend() {
+        legend = new HashMap();
+        legend.put(Block.Grass, Color.GREEN);
+        legend.put(Block.Path, Color.GRAY);
+        legend.put(Block.Hospital, Color.PINK);
+        legend.put(Block.House, Color.YELLOW);
+    }
```
Comment (N3moAhead) [inline src/com/biocollapse/view/MapPanel.java:119]: I think we also need to include the new Workplace block in here :)

**Review by N3moAhead — APPROVED** [review]
LGTM thanks for the change 👍🏻

**Review by Driemtax — APPROVED** [review]

**Pull request merged**